Lib/test/test_typing.py

```diff
@@ -43,6 +43,8 @@
 py_typing = import_helper.import_fresh_module('typing', blocked=['_typing'])
 c_typing = import_helper.import_fresh_module('typing', fresh=['_typing'])
 
+MODULE = ('test.test_typing', '__main__')[__name__ == '__main__']
```

Comment:
Does `__name__` not work?

Reply:
If I understood the line correctly, it's better to make logic more explicit:

```suggestion
MODULE = '__main__' if __name__ == '__main__' else 'test.test_typing'
```

Reply:
I've copied this pattern from another module I am familiar with: https://github.com/python/cpython/blob/83d544b9292870eb44f6fca37df0aa351c4ef83a/Lib/test/test_enum.py#L34

It has the exact same problem. I think that the initial motivation was that `__name__` has a hidden value inside. Who knows what is it without looking inside? But, with explicit names - it might be more readable.

But, I am fine with just `__name__` if you think it is more readable 🙂

Reply:
However, I support @zware that <code>self.assertEqual(SpecialAttrsT.\_\_module__, ***\_\_name__***)</code> would be better (if it works, of course).

Reply:
I deep dived into commit history of `test_enum` and found that this is most likely not a significant detail. So, I've removed `MODULE` and used plain `__name__`. I will also add a PR to simplify it there as well.

Thanks, @zware for noticing it!
And thanks to @arhadthedev for alternative solution 🙂